indico/modules/events/persons/controllers.py

```diff
@@ -257,47 +257,72 @@ def _process(self):
                                                has_predefined_affiliations=Affiliation.query.has_rows())
 
 
-class RHEmailEventPersons(RHManageEventBase):
+class RHEmailEventPersonsPreview(RHManageEventBase):
+    """Preview an email with EventPersons associated placeholders."""
+
+    @use_kwargs({
+        'body': fields.String(load_default=''),
+        'subject': fields.String(load_default=''),
+    })
+    def _process(self, body, subject):
+        person = self.event.persons[0]
+        email_body = replace_placeholders('event-persons-email', body, event=self.event, person=person)
+        email_subject = replace_placeholders('event-persons-email', subject, event=self.event, person=person)
+        tpl = get_template_module('events/persons/emails/custom_email.html', email_subject=email_subject,
+                                  email_body=email_body)
+        return {
+            'subject': tpl.get_subject(),
+            'body': tpl.get_body()
+        }
+
+
+class RHAPIEmailEventPersons(RHManageEventBase):
     """Send emails to selected EventPersons."""
 
     def _process_args(self):
-        self.no_account = request.args.get('no_account') == '1'
         RHManageEventBase._process_args(self)
+        self.no_account = request.args.get('no_account') == '1'
+        self.recipients = set(self._find_event_persons(request.args.getlist('person_id'),
```

Comment:
There's a reason why this was `request.form` and not `request.args`. ;)

When you select a LOT of people, you get a really long query string with `person_id=...&person_id=...&...` - which may end up being too long (and in any case it's super ugly in access logs). So using POST data for this is much cleaner.

Reply:
I'd also split this into two RHs (possibly with a base class having the `_process_args` since you need this both for sending the emails and getting the data for showing the form).

That way you can use POST for both cases (since the urls/endpoints are different) without having to use some extra arg to determine whether you want form metadata or send the emails.